specs/ceph_pool_usage_data.adoc

```diff
@@ -0,0 +1,175 @@
+utilization data
+
+* The cluster details should contain pool utilization data as well
+
+
+== Proposed change
+
+* Introduce a collectd plug-in which collects the pools utilization data from
+the underlying cluster and pushes the same graphite time series DB
+
+* Enhance the ceph cluster sync logic in Tendrl's ceph integration module to
+fetch process and update the pool utilization data to central store with
+cluster's pool inventory.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+* Update the `Pool` data model to include utilization related fields `used`,
+`total` and `PercentUsed`
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* The pool listing API `GET /cluster/:cluster_id/pool/:pool_id` should list the
+pool with additional fields related to utilization as below
+
+```
+Status: 200 OK
+
+{
+    'cluster_id': 'bc2c51e1-516d-45b8-8310-7caab2fa784c',
+    'min_size': '1',
+    'pg_num': '128',
+    'pool_id': '1',
+    'pool_name': 'testpool',
+    'utilization': {
+        'used': 0,
+        'total': 1024,
+        'percent_used': 0
+    }
+}
+```
+
+==== Notifications/Monitoring impact:
+
+* Add a collectd plug-in in monitoring module for getting the pool utilization
+data and pushing it to time series DB
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+* Tendrl's ceph integration module should synchronize the pool utilization data
+as well while cluster data getting synchronized
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+* The REST output for listing individual pools would be including additional
+utilization details
+
+=== Performance impact:
+
+* As confirmed by ceph team, the execution of `ceph df` command on leader mon is
+not resource intensive.
+
+* The collectd plug-in frequency should not be set to anything less than 5
+seconds as ceph cluser itself syncs data at this interval
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+* API module owner to make sure pool listing contains the utilization details
+as well and shown properly
+
+== Implementation:
+
+* Add a collectd plug-in to execute `ceph df` command to fetch the pool
+utilization data and push to time series database. The execution of `ceph df`
+command on a mon node is not heavy and as per suggestions from ceph team the
+connection to the mon could be kept open and every few seconds (not less than 5
+seconds as ceph itself syncs data at this interval) the pool utilization data
+could be pushed to time series DB
+
+* Make sure this plug-in gets activated only when monitoring is enabled for
+Tendrl and this plug-in should be enabled only on mon nodes of the cluster. The
+monitoring module changes to be done to activate the plug-in only on mon nodes.
+Also the ceph pool utilization plug-in should work on leader mon node only, to
+minimize the load on cluster
+
+* Enhance the ceph integration module's cluster data syncing logic to fetch the
```

Comment:
Could you add the implementation details suggested on ceph-devel, from your comment on the issue, here?